From 3312a89daf709c790f98388831befbfb03a81085 Mon Sep 17 00:00:00 2001
From: Joe Turner <jturner@freewave.com>
Date: Mon, 1 Aug 2022 22:54:19 -0500
Subject: [PATCH 1/3] Add device trees for fusion/huron devices

Manually applied patch for pcie/device tree changes to support
the devices.  Fixed rejected hunks.
---
 arch/arm/boot/dts/Makefile                   |   1 +
 arch/arm/boot/dts/imx6dl-fw_huron.dts        |  22 +
 arch/arm/boot/dts/imx6q-fw_huron.dts         |  26 +
 arch/arm/boot/dts/imx6qdl-fw_huron-base.dtsi | 506 +++++++++++++++++++
 arch/arm/boot/dts/imx6qdl-fw_huron.dtsi      | 149 ++++++
 arch/arm/boot/dts/imx6qdl.dtsi               |   5 +-
 6 files changed, 707 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx6dl-fw_huron.dts
 create mode 100644 arch/arm/boot/dts/imx6q-fw_huron.dts
 create mode 100644 arch/arm/boot/dts/imx6qdl-fw_huron-base.dtsi
 create mode 100644 arch/arm/boot/dts/imx6qdl-fw_huron.dtsi

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 1704e839868d..73097ddaa559 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -462,6 +462,7 @@ dtb-$(CONFIG_SOC_IMX53) += \
 	imx53-tx53-x13x.dtb \
 	imx53-usbarmory.dtb \
 	imx53-voipac-bsb.dtb
+#//IPS added fw_huron.dtbs
 dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6dl-alti6p.dtb \
 	imx6dl-apf6dev.dtb \
diff --git a/arch/arm/boot/dts/imx6dl-fw_huron.dts b/arch/arm/boot/dts/imx6dl-fw_huron.dts
new file mode 100644
index 000000000000..7cfe95d33e1e
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dl-fw_huron.dts
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2013 Freescale Semiconductor, Inc.
+ *
+ * Author: Fabio Estevam <fabio.estevam@freescale.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ */
+/dts-v1/;
+#include "imx6dl.dtsi"
+#include "imx6qdl-fw_huron.dtsi"
+
+/ {
+	model = "FreeWave Huron i.MX6 Solo/Dual Lite Board";
+	compatible = "freewave,imx6dl-fw_huron", "fsl,imx6dl";
+
+	memory {
+		reg = <0x10000000 0x40000000>;
+	};
+};
diff --git a/arch/arm/boot/dts/imx6q-fw_huron.dts b/arch/arm/boot/dts/imx6q-fw_huron.dts
new file mode 100644
index 000000000000..6cdc14250ff8
--- /dev/null
+++ b/arch/arm/boot/dts/imx6q-fw_huron.dts
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2013 Freescale Semiconductor, Inc.
+ *
+ * Author: Fabio Estevam <fabio.estevam@freescale.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ */
+/dts-v1/;
+#include "imx6q.dtsi"
+#include "imx6qdl-fw_huron.dtsi"
+
+/ {
+	model = "FreeWave Huron i.MX6 Quad Board";
+	compatible = "freewave,imx6q-fw_huron", "fsl,imx6q";
+
+	memory {
+		reg = <0x10000000 0x80000000>;
+	};
+};
+
+&sata {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx6qdl-fw_huron-base.dtsi b/arch/arm/boot/dts/imx6qdl-fw_huron-base.dtsi
new file mode 100644
index 000000000000..41ffe42064b7
--- /dev/null
+++ b/arch/arm/boot/dts/imx6qdl-fw_huron-base.dtsi
@@ -0,0 +1,506 @@
+/*
+ * Copyright 2013 Freescale Semiconductor, Inc.
+ *
+ * Author: Fabio Estevam <fabio.estevam@freescale.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ */
+
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	chosen {
+		stdout-path = &uart1;
+	};
+
+	regulators {
+		compatible = "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		reg_2p5v: regulator@0 {
+			compatible = "regulator-fixed";
+			reg = <0>;
+			regulator-name = "2P5V";
+			regulator-min-microvolt = <2500000>;
+			regulator-max-microvolt = <2500000>;
+			regulator-always-on;
+		};
+
+		reg_3p3v: regulator@1 {
+			compatible = "regulator-fixed";
+			reg = <1>;
+			regulator-name = "3P3V";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-always-on;
+		};
+	};
+
+	/* JJN
+	 sound {
+		compatible = "fsl,imx6-wandboard-sgtl5000",
+			     "fsl,imx-audio-sgtl5000";
+		model = "imx6-wandboard-sgtl5000";
+		ssi-controller = <&ssi1>;
+		audio-codec = <&codec>;
+		audio-routing =
+			"MIC_IN", "Mic Jack",
+			"Mic Jack", "Mic Bias",
+			"Headphone Jack", "HP_OUT";
+		mux-int-port = <1>;
+		mux-ext-port = <3>;
+	};
+
+	sound-spdif {
+		compatible = "fsl,imx-audio-spdif";
+		model = "imx-spdif";
+		spdif-controller = <&spdif>;
+		spdif-out;
+	};*/
+};
+
+&audmux {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_audmux>;
+	status = "okay";
+};
+
+&hdmi {
+	ddc-i2c-bus = <&i2c1>;
+	status = "okay";
+};
+
+&i2c1 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c1>;
+	status = "okay";
+};
+
+&i2c2 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c2>;
+	status = "okay";
+
+/* Huron: pmic needs further config to be used from here
+	pmic: pfuze100@08 {
+		compatible = "fsl,pfuze100";
+		reg = <0x08>;
+
+		regulators {
+			sw1a_reg: sw1ab {
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1875000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw1c_reg: sw1c {
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1875000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw2_reg: sw2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw3a_reg: sw3a {
+				regulator-min-microvolt = <400000>;
+				regulator-max-microvolt = <1975000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw3b_reg: sw3b {
+				regulator-min-microvolt = <400000>;
+				regulator-max-microvolt = <1975000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw4_reg: sw4 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <3300000>;
+			};
+
+			swbst_reg: swbst {
+				regulator-min-microvolt = <5000000>;
+				regulator-max-microvolt = <5150000>;
+			};
+
+			snvs_reg: vsnvs {
+				regulator-min-microvolt = <1000000>;
+				regulator-max-microvolt = <3000000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vref_reg: vrefddr {
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vgen1_reg: vgen1 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+			};
+
+			vgen2_reg: vgen2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+			};
+
+			vgen3_reg: vgen3 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+			};
+
+			vgen4_reg: vgen4 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen5_reg: vgen5 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen6_reg: vgen6 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+		};
+	}; */
+
+};
+
+&iomuxc {
+	pinctrl-names = "default";
+
+	imx6qdl-fw_huron {
+
+		pinctrl_audmux: audmuxgrp {
+			fsl,pins = <
+				MX6QDL_PAD_CSI0_DAT7__AUD3_RXD		0x130b0
+				MX6QDL_PAD_CSI0_DAT4__AUD3_TXC		0x130b0
+				MX6QDL_PAD_CSI0_DAT5__AUD3_TXD		0x110b0
+				MX6QDL_PAD_CSI0_DAT6__AUD3_TXFS		0x130b0
+			>;
+		};
+
+		pinctrl_enet: enetgrp {
+			fsl,pins = <
+				MX6QDL_PAD_ENET_MDIO__ENET_MDIO		0x1b0b0
+				MX6QDL_PAD_ENET_MDC__ENET_MDC		0x1b0b0
+				MX6QDL_PAD_RGMII_TXC__RGMII_TXC		0x1b030
+				MX6QDL_PAD_RGMII_TD0__RGMII_TD0		0x1b030
+				MX6QDL_PAD_RGMII_TD1__RGMII_TD1		0x1b030
+				MX6QDL_PAD_RGMII_TD2__RGMII_TD2		0x1b030
+				MX6QDL_PAD_RGMII_TD3__RGMII_TD3		0x1b030
+				MX6QDL_PAD_RGMII_TX_CTL__RGMII_TX_CTL	0x1b030
+				MX6QDL_PAD_ENET_REF_CLK__ENET_TX_CLK	0x1b0b0
+				MX6QDL_PAD_RGMII_RXC__RGMII_RXC		0x1b030
+				MX6QDL_PAD_RGMII_RD0__RGMII_RD0		0x1b030
+				MX6QDL_PAD_RGMII_RD1__RGMII_RD1		0x1b030
+				MX6QDL_PAD_RGMII_RD2__RGMII_RD2		0x1b030
+				MX6QDL_PAD_RGMII_RD3__RGMII_RD3		0x1b030
+				MX6QDL_PAD_RGMII_RX_CTL__RGMII_RX_CTL	0x1b030
+				MX6QDL_PAD_GPIO_16__ENET_REF_CLK	0x4001b0a8
+				MX6QDL_PAD_GPIO_6__ENET_IRQ		0x000b1
+			>;
+		};
+
+		pinctrl_i2c1: i2c1grp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D21__I2C1_SCL		0x4001b8b1
+				MX6QDL_PAD_EIM_D28__I2C1_SDA		0x4001b8b1
+			>;
+		};
+
+		pinctrl_i2c2: i2c2grp {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_COL3__I2C2_SCL		0x4001b8b1
+				MX6QDL_PAD_KEY_ROW3__I2C2_SDA		0x4001b8b1
+			>;
+		};
+
+		pinctrl_i2c3: i2c3grp {
+			fsl,pins = <
+				MX6QDL_PAD_GPIO_5__I2C3_SCL  		0x4001b8b1
+				MX6QDL_PAD_GPIO_16__I2C3_SDA 		0x4001b8b1
+			>;
+		};
+
+		pinctrl_spdif: spdifgrp {
+			fsl,pins = <
+				MX6QDL_PAD_ENET_RXD0__SPDIF_OUT		0x1b0b0
+			>;
+		};
+
+		pinctrl_ecspi3: ecspi3grp {
+			fsl,pins = <
+				MX6QDL_PAD_DISP0_DAT2__ECSPI3_MISO	0x100b1
+				MX6QDL_PAD_DISP0_DAT1__ECSPI3_MOSI	0x100b1
+				MX6QDL_PAD_DISP0_DAT0__ECSPI3_SCLK	0x100b1
+	/* use CS as GPIO	MX6QDL_PAD_DISP0_DAT3__ECSPI3_SS0	0x1b0b0 */
+				MX6QDL_PAD_DISP0_DAT3__GPIO4_IO24	0x1b0b0
+			>;
+		};
+
+		pinctrl_uart1: uart1grp {
+			fsl,pins = <
+				MX6QDL_PAD_CSI0_DAT10__UART1_TX_DATA	0x1b0b1
+				MX6QDL_PAD_CSI0_DAT11__UART1_RX_DATA	0x1b0b1
+			       /*
+				MX6QDL_PAD_EIM_D19__UART1_CTS_B		0x1b0b1
+				MX6QDL_PAD_EIM_D20__UART1_RTS_B		0x1b0b1
+                              */
+				MX6QDL_PAD_EIM_D19__GPIO3_IO19		0x1b0b0
+				MX6QDL_PAD_EIM_D20__GPIO3_IO20		0x1b0b0
+			>;
+		};
+
+		pinctrl_uart2: uart2grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD3_DAT4__UART2_RX_DATA	0x1b0b1
+				MX6QDL_PAD_SD3_DAT5__UART2_TX_DATA	0x1b0b1
+			>;
+		};
+
+/*
+	pinctrl_uart3: uart3grp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D24__UART3_TX_DATA	0x1b0b1
+				MX6QDL_PAD_EIM_D25__UART3_RX_DATA	0x1b0b1
+				MX6QDL_PAD_EIM_D23__UART3_CTS_B		0x1b0b1
+				MX6QDL_PAD_EIM_EB3__UART3_RTS_B		0x1b0b1
+			>;
+		};
+*/
+
+		pinctrl_uart4: uart4grp {
+			fsl,pins = <
+				MX6QDL_PAD_CSI0_DAT12__UART4_TX_DATA	0x1b0b1
+				MX6QDL_PAD_CSI0_DAT13__UART4_RX_DATA	0x1b0b1
+				MX6QDL_PAD_CSI0_DAT17__UART4_CTS_B	0x1b0b1
+				MX6QDL_PAD_CSI0_DAT16__UART4_RTS_B	0x1b0b1
+			>;
+		};
+
+		pinctrl_pcie_reset: pciegrp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D31__GPIO3_IO31 		0x130b0
+			>;
+		};
+
+
+		pinctrl_usbotg: usbotggrp {
+			fsl,pins = <
+				MX6QDL_PAD_ENET_RX_ER__USB_OTG_ID	0x17059
+			>;
+		};
+
+/*
+		pinctrl_usdhc1: usdhc1grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD1_CMD__SD1_CMD		0x17059
+				MX6QDL_PAD_SD1_CLK__SD1_CLK		0x10059
+				MX6QDL_PAD_SD1_DAT0__SD1_DATA0		0x17059
+				MX6QDL_PAD_SD1_DAT1__SD1_DATA1		0x17059
+				MX6QDL_PAD_SD1_DAT2__SD1_DATA2		0x17059
+//				MX6QDL_PAD_SD1_DAT3__SD1_DATA3		0x17059
+			>;
+		};
+*/
+
+		pinctrl_usdhc2: usdhc2grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD2_CMD__SD2_CMD		0x17059
+				MX6QDL_PAD_SD2_CLK__SD2_CLK		0x10059
+				MX6QDL_PAD_SD2_DAT0__SD2_DATA0		0x17059
+				MX6QDL_PAD_SD2_DAT1__SD2_DATA1		0x17059
+				MX6QDL_PAD_SD2_DAT2__SD2_DATA2		0x17059
+				MX6QDL_PAD_SD2_DAT3__SD2_DATA3		0x17059
+			>;
+		};
+
+		pinctrl_usdhc3: usdhc3grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD3_CMD__SD3_CMD		0x17059
+				MX6QDL_PAD_SD3_CLK__SD3_CLK		0x10059
+				MX6QDL_PAD_SD3_DAT0__SD3_DATA0		0x17059
+				MX6QDL_PAD_SD3_DAT1__SD3_DATA1		0x17059
+				MX6QDL_PAD_SD3_DAT2__SD3_DATA2		0x17059
+				MX6QDL_PAD_SD3_DAT3__SD3_DATA3		0x17059
+			>;
+		};
+
+		pinctrl_usdhc4: usdhc4grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD4_CMD__SD4_CMD		0x17059
+				MX6QDL_PAD_SD4_CLK__SD4_CLK		0x10059
+				MX6QDL_PAD_SD4_DAT0__SD4_DATA0		0x17059
+				MX6QDL_PAD_SD4_DAT1__SD4_DATA1		0x17059
+				MX6QDL_PAD_SD4_DAT2__SD4_DATA2		0x17059
+				MX6QDL_PAD_SD4_DAT3__SD4_DATA3		0x17059
+				MX6QDL_PAD_SD4_DAT4__SD4_DATA4		0x17059
+				MX6QDL_PAD_SD4_DAT5__SD4_DATA5		0x17059
+				MX6QDL_PAD_SD4_DAT6__SD4_DATA6		0x17059
+				MX6QDL_PAD_SD4_DAT7__SD4_DATA7		0x17059
+			>;
+		};
+		pinctrl_gpio_leds: gpioledsgrp {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_COL0__GPIO4_IO06		0x1b0b0	/* WDOG_INP (GPIO102) */
+			>;
+		};
+		pinctrl_wdog: wdoggrp {
+			fsl,pins = <
+				MX6QDL_PAD_SD1_DAT3__WDOG2_B		0x1b0b0	/* WDOG output for HW reset */
+
+			>;
+		};
+
+	};
+};
+
+&fec {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_enet>;
+	phy-mode = "rgmii";
+	phy-reset-gpios = <&gpio3 29 GPIO_ACTIVE_LOW>;
+	interrupts-extended = <&gpio1 6 IRQ_TYPE_LEVEL_HIGH>,
+			      <&intc 0 119 IRQ_TYPE_LEVEL_HIGH>;
+	fsl,err006687-workaround-present;
+	status = "okay";
+};
+
+&pcie  {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pcie_reset>;
+	reset-gpio = <&gpio3 31 GPIO_ACTIVE_HIGH>;
+	status = "okay";
+};
+
+&ecspi3 {
+	cs-gpios = <&gpio4 24 0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi3>;
+	status = "okay";
+
+	flash: spi-flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "macronix,mx25v8035f", "issi,is25wp080d", "jedec,spi-nor";
+		spi-max-frequency = <20000000>;
+		reg = <0>;
+		partitions@0 {
+			label = "Partition1";
+			reg = <0x0 0x100000>;
+		};
+	};
+};
+
+&ssi1 {
+	status = "okay";
+};
+
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1>;
+	/*uart-has-rtscts;*/ /* this means using UART block to control RTS,CTS. since we use gpio for RTS, CTS, comment it out */
+	status = "okay";
+	cts-gpios = <&gpio3 19 GPIO_ACTIVE_LOW>;
+	rts-gpios = <&gpio3 20 GPIO_ACTIVE_LOW>;
+};
+
+&uart2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart2>;
+	status = "okay";
+};
+
+/*
+&uart3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart3>;
+	uart-has-rtscts;
+	status = "okay";
+};
+*/
+
+&uart4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart4>;
+	uart-has-rtscts;
+	status = "okay";
+};
+
+&usbh1 {
+	status = "okay";
+};
+
+&usbotg {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usbotg>;
+	disable-over-current;
+	dr_mode = "peripheral";
+	status = "okay";
+};
+
+/*
+&usdhc1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc1>;
+	bus-width = <4>;
+	cd-gpios = <&gpio1 2 GPIO_ACTIVE_LOW>;
+	status = "okay";
+};
+*/
+
+&usdhc2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc2>;
+	status = "okay";
+};
+
+&usdhc3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc3>;
+	cd-gpios = <&gpio3 9 GPIO_ACTIVE_LOW>;
+	status = "okay";
+};
+
+&usdhc4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc4>;
+	bus-width = <8>;
+	non-removable;
+	status = "okay";
+};
+&wdog1 {
+	status = "disabled";
+};
+
+&wdog2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_wdog>;
+	fsl,ext-reset-output;
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx6qdl-fw_huron.dtsi b/arch/arm/boot/dts/imx6qdl-fw_huron.dtsi
new file mode 100644
index 000000000000..ce74fb897162
--- /dev/null
+++ b/arch/arm/boot/dts/imx6qdl-fw_huron.dtsi
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2013 Freescale Semiconductor, Inc.
+ *
+ * Author: Fabio Estevam <fabio.estevam@freescale.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ */
+
+#include "imx6qdl-fw_huron-base.dtsi"
+
+/* Huron GPIOs
+* RGMII_nRST
+*   RGMII_nRST			:  MX6QDL_PAD_EIM_D29__GPIO3_IO29 RGMII_nRST (GPIO93)
+*
+* UART1
+*   RS485nRS232	        	:  PAD_ENET_RXD1__GPIO1_IO26 (GPIO26)
+*   FAST MODE  			:  EIM_EB0 - MX6QDL_PAD_EIM_EB0__GPIO2_IO28 (GPIO60) Active high
+*   RS232_INVALID 		:  EIM_EB1 - MX6QDL_PAD_EIM_EB1__GPIO2_IO29 (GPIO61) Input
+*   485_HALF_nFULL		:  EIM_EB2 - MX6QDL_PAD_EIM_EB2__GPIO2_IO30 (GPIO62)
+*   UART1_ENABLE	       	:  PAD_EIM_EB3__GPIO2_IO31 (GPIO63) Active high
+*   UART1_DCD (GPIO)		:  MX6QDL_PAD_EIM_D23__GPIO3_IO23 (GPIO87)
+*   UART1_DTR (GPIO)		:  MX6QDL_PAD_EIM_D24__GPIO3_IO24 (GPIO88)
+*
+* USB and PERIPHERAL POWER
+*   USB_HOST_PWR_ENABLE    	:  PAD_GPIO_1__GPIO1_IO01 (MX6QDL_PAD_GPIO_1__USB_OTG_ID) (GPIO1) Active HIGH
+*   USB_OTG_PWR_EN		:  NANDF_CS1 - MX6QDL_PAD_NANDF_CS1__GPIO6_IO14 (GPIO174) Active HIGH
+*   3V3_PER_PWR_EN		:  CSI0_DATA_EN - MX6QDL_PAD_CSI0_DATA_EN__GPIO5_IO20 (GPIO148) Active HIGH
+*   USB_HUB_nRST		:  GPIO_17 - MX6QDL_PAD_GPIO_17__GPIO7_IO12 (GPIO204) Active LOW
+*
+* ZUMLINK CONTROL
+*   ZL_MOD_PWR_EN 		:  DI0_DISP_CLK - MX6QDL_PAD_DI0_DISP_CLK__GPIO4_IO16 (GPIO112) (output)
+*   ZL_MOD_CD			:  DI0_PIN3 - MX6QDL_PAD_DI0_PIN3__GPIO4_IO19 (GPIO115) (input)
+*   ZL_MOD_nRST			:  DI0_PIN15 - MX6QDL_PAD_DI0_PIN15__GPIO4_IO17 (GPIO113) (output)
+*   ZL_MOD_nINT			:  DI0_PIN2 - MX6QDL_PAD_DI0_PIN2__GPIO4_IO18 (GPIO114)
+*   ZL_MOD_DTR			:  DI0_PIN4 - MX6QDL_PAD_DI0_PIN4__GPIO4_IO20  (GPIO116)
+*
+* LEDS
+*   UP_RX_RED			:  DISP0_DAT10 - MX6QDL_PAD_DISP0_DAT10__GPIO4_IO31 (GPIO127)
+*   UP_WLAN_RED			:  DISP0_DAT8 - MX6QDL_PAD_DISP0_DAT9__GPIO4_IO30 (GPIO126)
+*   UP_RX_GREEN			:  DISP0_DAT8 - MX6QDL_PAD_DISP0_DAT8__GPIO4_IO29 (GPIO125)
+*   UP_TX_RED			:  DISP0_DAT7 - MX6QDL_PAD_DISP0_DAT7__GPIO4_IO28 (GPIO124)
+*   UP_HEART_BEAT		:  DISP0_DAT11 - MX6QDL_PAD_DISP0_DAT11__GPIO5_IO05 (GPIO133)
+*   UP_CD_GREEN			:  DISP0_DAT12 - MX6QDL_PAD_DISP0_DAT12__GPIO5_IO06 (GPIO134)
+*   UP_WLAN_nGREEN		:  DISP0_DAT13 - MX6QDL_PAD_DISP0_DAT13__GPIO5_IO07 (GPIO135) Active LOW
+*   UP_BT_RED			:  DISP0_DAT14 - MX6QDL_PAD_DISP0_DAT14__GPIO5_IO08 (GPIO136)
+*   UP_BT_nGREEN		:  DISP0_DAT15 - MX6QDL_PAD_DISP0_DAT15__GPIO5_IO09 (GPIO137) Active LOW
+*   UP_CD_RED			:  DISP0_DAT16 - MX6QDL_PAD_DISP0_DAT16__GPIO5_IO10 (GPIO138)
+*   COM1_LED_GR_L		:  DISP0_DAT17 - MX6QDL_PAD_DISP0_DAT17__GPIO5_IO11 (GPIO139)
+*   COM1_LED_GR_R		:  DISP0_DAT18 - MX6QDL_PAD_DISP0_DAT18__GPIO5_IO12 (GPIO140)
+*
+* WDOG
+*   WDOG_INP			:  KEY_COL0 - MX6QDL_PAD_KEY_COL0__GPIO4_IO06 (GPIO102)
+*   WDOG_nEN_DIS		:  KEY_ROW0 - MX6QDL_PAD_KEY_ROW0__GPIO4_IO07 (GPIO103)
+*
+* RF HW ENABLE/KILL
+*   BT_nDISABLE			:  CSI0_VSYNC - MX6QDL_PAD_CSI0_VSYNC__GPIO5_IO21 (GPIO149)
+*   WLAN_nDISABLE		:  GPIO_19 - MX6QDL_PAD_GPIO_19__GPIO4_IO05 (GPIO101)
+*
+* SPI FLASH DISCRETE LINES
+*   SPI_NOR_nWP			:  ECSPI3_WP# - MX6QDL_PAD_DISP0_DAT4__GPIO4_IO25 (GPIO121)
+*   SPI_NOR_nHOLD		:  ECSPI3_HOLD# - MX6QDL_PAD_DISP0_DAT5__GPIO4_IO26 (GPIO122)
+*
+* POWER ON/OFF MGMT
+*   SOFT_SHDN			:  GPIO1_3 - MX6QDL_PAD_GPIO_3__GPIO1_IO03 (GPIO3) soft shutdown output
+*   PFI_IMX6			:  GPIO1_9 - MX6QDL_PAD_GPIO_9__GPIO1_IO09 (GPIO9) PowerFail/Down input
+*
+*/
+
+/ {
+
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_gpio_leds>;
+		/* map external watchdog keepalive to heartbeat trigger */
+		led0: heartbeat {
+			label = "heartbeat";
+			gpios = <&gpio4 6 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+			linux,default-trigger = "heartbeat";
+		};
+
+	};
+
+};
+
+&iomuxc {
+
+	pinctrl-0 = <&pinctrl_hog>;
+	imx6qdl-wandboard {
+		pinctrl_hog: hoggrp {
+			fsl,pins = <
+/* SD card CD */
+				MX6QDL_PAD_EIM_DA9__GPIO3_IO09		0x80000000	/* uSDHC3 CD */
+/* RGMII_nRST */
+				MX6QDL_PAD_EIM_D29__GPIO3_IO29		0x80000000	/* RGMII_nRST */
+/* UART1 discrete lines */
+				MX6QDL_PAD_ENET_RXD1__GPIO1_IO26	0x1b0b0		/* RS485_nRS232  */
+				MX6QDL_PAD_EIM_EB0__GPIO2_IO28 		0x1b0b0  	/* FAST MODE ) */
+				MX6QDL_PAD_EIM_EB1__GPIO2_IO29 		0x1b0b0		/* RS232_INVALID - EIM_EB1 */
+				MX6QDL_PAD_EIM_EB2__GPIO2_IO30 		0x1b0b0		/* 485_HALF_nFULL - EIM_EB2 */
+				MX6QDL_PAD_EIM_EB3__GPIO2_IO31		0x1b0b0		/* UART1_ENABLE */
+				MX6QDL_PAD_EIM_D23__GPIO3_IO23		0x1b0b0		/* UART1_DCD (GPIO) */
+				MX6QDL_PAD_EIM_D24__GPIO3_IO24		0x1b0b0		/* UART1_DTR (GPIO) */
+/* Power enable lines */
+				MX6QDL_PAD_GPIO_1__GPIO1_IO01		0x1b0b0		/* USB_HOST_PWR_ENABLE  */
+				MX6QDL_PAD_NANDF_CS1__GPIO6_IO14 	0x1b0b0		/* USB_OTG_PWR_ENABLE - NANDF_CS1 */
+				MX6QDL_PAD_CSI0_DATA_EN__GPIO5_IO20 	0x1b0b0		/* 3V3_PER_PWR_EN - CSI0_DATA_EN */
+				MX6QDL_PAD_GPIO_17__GPIO7_IO12 		0x1b0b0		/* USB_HUB_nRST - GPIO_17 */
+//JJN +5 ZumLink interface
+				MX6QDL_PAD_DI0_DISP_CLK__GPIO4_IO16	0x1b0b0		/* ZL_MOD_PWR_EN - power enable (active high) */
+				MX6QDL_PAD_DI0_PIN3__GPIO4_IO19		0x1b0b0		/* ZL_MOD_CD (input) */
+				MX6QDL_PAD_DI0_PIN15__GPIO4_IO17	0x1b0b0		/* ZL_MOD_nRST (output) */
+				MX6QDL_PAD_DI0_PIN2__GPIO4_IO18		0x1b0b0		/* ZL_MOD_nINT (input) */
+				MX6QDL_PAD_DI0_PIN4__GPIO4_IO20		0x1b0b0		/* ZL_MOD_DTR (output */
+/* Huron LEDs */
+				MX6QDL_PAD_DISP0_DAT10__GPIO4_IO31	0x0b0b0		/* LED - UP_RX_RED */
+				MX6QDL_PAD_DISP0_DAT9__GPIO4_IO30	0x1b0b0		/* LED - UP_WLAN_RED */
+				MX6QDL_PAD_DISP0_DAT8__GPIO4_IO29	0x1b0b0		/* LED - UP_RX_GREEN */
+				MX6QDL_PAD_DISP0_DAT7__GPIO4_IO28 	0x1b0b0		/* LED - UP_TX_RED */
+				MX6QDL_PAD_DISP0_DAT11__GPIO5_IO05 	0x1b0b0		/* LED - UP_HEART_BEAT */
+				MX6QDL_PAD_DISP0_DAT12__GPIO5_IO06	0x1b0b0		/* LED - UP_CD_GREEN */
+				MX6QDL_PAD_DISP0_DAT13__GPIO5_IO07	0x1b0b0		/* LED - UP_WLAN_nGREEN */
+				MX6QDL_PAD_DISP0_DAT14__GPIO5_IO08	0x1b0b0		/* LED - UP_BT_RED */
+				MX6QDL_PAD_DISP0_DAT15__GPIO5_IO09	0x1b0b0		/* LED - UP_BT_nGREEN */
+				MX6QDL_PAD_DISP0_DAT16__GPIO5_IO10 	0x1b0b0		/* LED - UP_CD_RED */
+				MX6QDL_PAD_DISP0_DAT17__GPIO5_IO11	0x1b0b0		/* LED - COM1_LED_GR_L */
+				MX6QDL_PAD_DISP0_DAT18__GPIO5_IO12	0x1b0b0		/* LED - COM!_LED_GR_R */
+/* moved External WDOG_INP to LED Group for control through heartbeat trigger */
+//				MX6QDL_PAD_KEY_COL0__GPIO4_IO06		0x1b0b0		/* WDOG_INP */
+				MX6QDL_PAD_KEY_ROW0__GPIO4_IO07		0x1b0b0		/* WDOG_nEN_DIS */
+/* RF HW Enable/Kill */
+				MX6QDL_PAD_CSI0_VSYNC__GPIO5_IO21 	0x1b0b0		/* BT_nDISABLE - CSI0_VSYNC */
+				MX6QDL_PAD_GPIO_19__GPIO4_IO05 		0x1b0b0		/* WLAN_nDISABLE - GPIO_19 */
+/* SPI-NOR FLASH nWP and nHOLD GPIOS (active low) */
+				MX6QDL_PAD_DISP0_DAT4__GPIO4_IO25	0x1b0b0		/* SPI_NOR_nWP - ECSPI3_WP# */
+				MX6QDL_PAD_DISP0_DAT5__GPIO4_IO26 	0x1b0b0		/* SPI_NOR_nHOLD - ECSPI3_HOLD# */
+/* Power ON/OFF Management */
+				MX6QDL_PAD_GPIO_3__GPIO1_IO03		0x1b0b0		/* soft shutdown output */
+				MX6QDL_PAD_GPIO_9__GPIO1_IO09		0x1b0b0		/* PowerFail/Down indicator */
+			>;
+		};
+	};
+
+
+};
diff --git a/arch/arm/boot/dts/imx6qdl.dtsi b/arch/arm/boot/dts/imx6qdl.dtsi
index 652feff33496..d47e9a282233 100644
--- a/arch/arm/boot/dts/imx6qdl.dtsi
+++ b/arch/arm/boot/dts/imx6qdl.dtsi
@@ -1281,8 +1281,9 @@ uart4: serial@21f0000 {
 				clocks = <&clks IMX6QDL_CLK_UART_IPG>,
 					 <&clks IMX6QDL_CLK_UART_SERIAL>;
 				clock-names = "ipg", "per";
-				dmas = <&sdma 31 4 0>, <&sdma 32 4 0>;
-				dma-names = "rx", "tx";
+//IPS -2, Disable DMA
+//				dmas = <&sdma 31 4 0>, <&sdma 32 4 0>;
+//				dma-names = "rx", "tx";
 				status = "disabled";
 			};
 
-- 
2.34.1

